elastalert/ruletypes.py

```diff
@@ -58,9 +58,16 @@ def add_match(self, event):
         if ts in event:
             event[ts] = dt_to_ts(event[ts])
 
-        if self.rules.get('compound_query_key') and '_source' in event:
-            event.update({key: lookup_es_key(event['_source'], key) for key in self.rules['compound_query_key']})
-
+        query_key = self.rules.get('query_key')
+        event_key = event.get('key')
+        # make sure events have the query_key values in the query_key fields
+        # this is only relevant when we have a "key" field, which is Elastalert's
+        # way to provide information when lacking document data
+        if query_key is not None and event_key is not None:
+            split_query_key = [x.strip() for x in query_key.split(',')]
```

Comment:
`split_query_key = self.rules.get('compound_query_key') if 'compound_query_key' in self.rules else [self.rules.get('query_key')]` should work; I think you can assume it will be ordered the same.